File: 백준/Silver/1543. 문서 검색/문서 검색.py
import sys
input = sys.stdin.readline
doc = input().strip()
find = input().strip()
ans = 0
i = 0
while i <= len(doc)-len(find):
    if doc[i:i+len(find)] == find:
        ans += 1
        i += len(find)
    else:
        i += 1
print(ans)

File: 백준/Silver/1543. 문서 검색/문서 검색.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
const doc = input[0];
const find = input[1];
let ans = 0;
let i = 0;

while (i <= doc.length - find.length) {
  if (doc.slice(i, i+find.length) === find) {
    ans++;
    i += find.length;
  } else {
    i++;
  }
}

console.log(ans);